wsmaster/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/jpa/JpaSshDao.java

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.ssh.server.jpa;
+
+import com.google.inject.persist.Transactional;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.jdbc.jpa.DuplicateKeyException;
+import org.eclipse.che.api.ssh.server.model.impl.SshPairImpl;
+import org.eclipse.che.api.ssh.server.spi.SshDao;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * JPA based implementation of {@link SshDao}.
+ *
+ * @author Mihail Kuznyetsov
+ */
+@Singleton
+public class JpaSshDao implements SshDao {
+
+    @Inject
+    private Provider<EntityManager> managerProvider;
+
+    @Override
+    public void create(SshPairImpl sshPair) throws ServerException, ConflictException {
+        requireNonNull(sshPair);
+        try {
+            doCreate(sshPair);
+        } catch (DuplicateKeyException e) {
+            throw new ConflictException(String.format("Ssh pair with service '%s' and name '%s' already exists",
+                                                      sshPair.getService(),
+                                                      sshPair.getName()));
+        } catch (RuntimeException e) {
+            throw new ServerException(e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public List<SshPairImpl> get(String owner, String service) throws ServerException {
+        requireNonNull(owner);
+        requireNonNull(service);
+        try {
+            EntityManager manager = managerProvider.get();
+            List<SshPairImpl> resultList = manager.createQuery("SELECT pair " +
+                                                               "FROM SshKeyPair pair " +
+                                                               "WHERE pair.owner = :owner " +
+                                                               "  AND pair.service = :service", SshPairImpl.class)
```

Comment:
actually, it is not needed, since it consist of primary key parts, and their order allows to reuse primary key for indexing